Add fault diagram to ASB, RabbitMQ, and SQS topology diagram that explains the flow of messages for errors (#5501)
MassTransit/MassTransit: doc/content/3.documentation/6.transports/3.azure-service-bus.md

## Broker Topology

> In this example topology, two commands and events are used.

Reviewer: ```suggestion
Two commands and events are used in this example.
```

> First, the event contracts that are supported by an endpoint that receives files from a customer.

Reviewer: ```suggestion
These are the event contracts for a consumer that receives files from a customer:
```

```csharp
namespace Acme;

public interface FileReceived
{
    Guid FileId { get; }
    DateTime Timestamp { get; }
    Uri Location { get; }
}

public interface CustomerDataReceived
{
    DateTime Timestamp { get; }
    string CustomerId { get; }
    string SourceAddress { get; }
    Uri Location { get; }
}
```

> Second, the command contract for processing a file that was received.

Reviewer: ```suggestion
Here is the command contract for processing a file that was received.
```

```csharp
namespace Acme;

public interface ProcessFile
{
    Guid FileId { get; }
    Uri Location { get; }
}
```

> The above contracts are used by the consumers to receive messages. From a publishing or sending perspective, two classes are created by the event producer and
> the command sender which implement these interfaces.

Reviewer: ```suggestion
The above contracts are used by the consumers to receive messages. From a publishing or sending perspective, two classes are created by the event producer and the command sender which implement these interfaces.
```

```csharp
namespace Acme;

public record FileReceivedEvent :
    FileReceived,
    CustomerDataReceived
{
    public Guid FileId { get; init; }
    public DateTime Timestamp { get; init; }
    public Uri Location { get; init; }
    public string CustomerId { get; init; }
    public string SourceAddress { get; init; }
}
```

> And the command class.

Reviewer: ```suggestion
And the command class:
```

```csharp
namespace Acme;

public record ProcessFileCommand :
    ProcessFile
{
    public Guid FileId { get; init; }   
    public Uri Location { get; init; }
}
```

> The consumers for these message contracts are as below.

Reviewer: ```suggestion
The consumers for these message contracts are shown below:
```

```csharp
namespace Acme;

class FileReceivedConsumer :
    IConsumer<FileReceived>
{
}

class CustomerAuditConsumer :
    IConsumer<CustomerDataReceived>
{
}

class ProcessFileConsumer :
    IConsumer<ProcessFile>
{
}
```

:::alert{type="info"}
The broker topology can be customized using the [topology API](/documentation/configuration/topology).
:::

### Send

> These are the exchanges and queues for above mentioned example topology for a Send:

Reviewer: ```suggestion
These are the topics and queues for the example above when Sending a message:
```

![Send topology for Azure Service Bus](/azure-topology-send.svg)

### Publish

> These are the exchanges and queues for above mentioned example topology for a Publish of a [polymorphic message that uses inheritance](/documentation/concepts/messages#message-inheritance):

Reviewer: ```suggestion
These are the topics and queues for the example above when Publishing a [polymorphic message that uses inheritance](/documentation/concepts/messages#message-inheritance):
```

![Publish topology for Azure Service Bus](/azure-topology-publish.svg)

### Fault

> These are the exchanges and queues used when messages fail. The failing message gets forwarded to an `_error` queue by default. The following diagram shows which Exchanges and Queues are used when a message fails to be processed and is deadlettered for the above mentioned example topology, as well as how fault result messages are sent.

Reviewer: ```suggestion
These are the topics and queues used when messages fail. The failing message gets forwarded to an `_error` queue by default. The following diagram shows which topics and queues are used when a message fails to be processed and is deadlettered for the example above.
```
